src/DeriveHasField.hs

```diff
@@ -24,14 +25,53 @@ deriveHasFieldWith fieldModifier = makeDeriveHasField fieldModifier <=< reifyDat
 
 deriveHasField :: Name -> DecsQ
 deriveHasField name = do
+  (datatypeInfo, constructorInfo) <- getSingleDataConstructorInfo name
+  fieldNames <- getRecordConstructorFieldNames constructorInfo
+  let prefix = lowerFirst $ nameBase constructorInfo.constructorName
+      dropPrefix input = fromMaybe input $ stripPrefix prefix input
+  validateFieldNames Assumed prefix fieldNames
```

Comment:
This checks that all field names match the data constructors. We "assume" the prefix in this case because it is just the `lowerFirst $ nameBase constructorInfo.constructorName`